File: backend/src/main/java/com/backend/shared/utils/MaskUtils.java
package com.backend.shared.utils;

public class MaskUtils {
    public static String maskcpf(String cpf) {
        if (cpf == null || cpf.length() != 11)
            return cpf;
        return "***.***.***-" + cpf.substring(9);
    }

    public static String maskTelephone(String telephone) {
        if (telephone == null || telephone.length() != 11)
            return telephone;
        return telephone.substring(0, 2) + " *****-" + telephone.substring(7);
    }

    public static String maskEmail(String email) {
        if (email == null || !email.contains("@"))
            return email;

        String[] parts = email.split("@");
        String nome = parts[0];

        String masked = nome.length() <= 2
                ? "***"
                : nome.substring(0, 2) + "***";

        return masked + "@" + parts[1];
    }
}

File: frontend/src/utils/maskUtils.js
export const MaskUtils = {
  maskCpf: (cpf) => {
    if (!cpf || cpf.length !== 11) return cpf;
    return `***.***.***-${cpf.substring(9)}`;
  },

  maskTelephone: (telephone) => {
    if (!telephone || telephone.length !== 11) return telephone;
    return `${telephone.substring(0, 2)} *****-${telephone.substring(7)}`;
  },

  maskEmail: (email) => {
    if (!email || !email.includes("@")) return email;

    const [nome, dominio] = email.split("@");
    const maskedNome = nome.length <= 2 ? "***" : `${nome.substring(0, 2)}***`;

    return `${maskedNome}@${dominio}`;
  },
};
